doc/modules/mod_event_pusher_push.md

```diff
@@ -1,57 +1,163 @@
-### Module Description
+## Module Description
 
 This module is a backend of [mod_event_pusher] that implements [XEP-0357: Push Notifications](https://xmpp.org/extensions/xep-0357.html).
-It enables a service that notifies `PubSub` of a user's choice about every message that they could miss while being offline.
-There are two control stanzas that the client can send to this module: `enable` and `disable`.
-The `enable` stanza enables push notifications and forwards them to a specified `PubSub` node.
-This stanza may also contain an optional `Data Form` that will be added to each and every notification to `PubSub` node as `publish-options`.
-Please be sure to provide all form fields required by the specified `PubSub` node.
-Any publish error may result in disabling push notifications to this node.
+It provides push notification data to the service that delivers actual notification to a client device.
 
-### Options
+We've prepared [a detailed tutorial](../user-guide/Push-notifications.md) about a proper push notifications setup both on a client and a server side.
+
+Please make sure that clients provide all form fields required by the specified `PubSub` node.
+Some publish errors may result in disabling push notifications for the specific device until it attempts to enable them again.
+
+## Options
 
 * **backend** (atom, default: `mnesia`) - Backend to use for storing the registrations.
  Possible options are `mnesia` and `rdbms`.
 * **wpool** (list, default: `[]`) - List of options that will be passed to the `worker_pool` library that handles all the requests.
  Please refer to the [Project Site](https://github.com/inaka/worker_pool) for more details.
 * **plugin_module** (atom, default: `mod_event_pusher_push_plugin_defaults`) - module implementing `mod_event_pusher_push_plugin` behaviour,
   used for dynamic configuration of push notifications. Read more about it [here](#plugin-module)
+* **virtual_pubsub_hosts** (list of strings, default: `[]`) - a list of "simulated" Publish-Subscribe domains. See [relevant section](#virtual-pubsub-hosts) for more details. You may use `@HOST@` pattern in the domain name. It will automatically be replaced by a respective XMPP domain (e.g. `localhost`)
 
-### Plugin module
+## Virtual PubSub hosts
 
-A plugin module handles dynamic configuration of push notifications. It implements `mod_event_pusher_push_plugin` behaviour which
-requires two callbacks:
+If a notification is published to one of the configured domains, the internal push notifications hook is executed in MongooseIM instead of the XEP-0357 typical behaviour.
```

Comment:
```suggestion
If a notification is published to one of the configured domains, the internal push notification hook is executed in MongooseIM instead of the XEP-0357 typical behaviour.
```